File: algorithm/src/cn/beijing/algorithm/ch01/java/BubbleSortComplex.java
/**
 * 
 */
package cn.beijing.algorithm.ch01.java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import cn.beijing.algorithm.ch01.pojo.Student;

/**
 * @author tsukai
 *
 */
public class BubbleSortComplex {
	public static void main(String[] args) throws IOException {
		Student a[] = new Student[10];
		Student t;
		String str;
		int i,j,n;
		System.out.println("please input the num that needs to sort:");
		BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
		str = reader.readLine();
		n = Integer.parseInt(str);
		for(i = 0;i < n;i++){
			str = reader.readLine();
			Student stu = new Student();
			stu.setName("name"+i);
			stu.setSocre(Integer.parseInt(str));
			a[i] = stu;
		}
		
		//sort
		for(i = 1;i <= n - 1;i++){
			for(j = 0;j < n - i;j++){
				if(a[j].getSocre() < a[j+1].getSocre()){
					t = a[j];
					a[j] = a[j+1];
					a[j+1] = t;
				}
			}
		}
		
		for (i = 0; i < n; i++) {
			System.out.println(a[i]);
		}
		
	}
}

File: algorithm/src/cn/beijing/algorithm/ch01/c/bubble_sort_complex.c
#include<stdio.h>

struct student
{
	char name[10];
	char score;
};

int main()
{
	struct student a[10] ,t;
	int i,j,n;
	printf("please input the num that needs to sort:\n");
	scanf("%d",&n);
	for(i = 0;i < n;i++){
		scnaf("%s %d",a[i].name,&a[i].score);
	}
	
	for(i = 1;i <= n - 1;i++){
		for(j = 0;j < n - i;j++){
			if(a[j].score < a[j+1].score){
				t = a[j];
				a[j] = a[j+1];
				a[j+1] = t;
			}
		}	
	}	
	
	for(i = 0;i < n;i++){
		printf("%s %d\n",a[i].name,a[i].score);
	}
	getchar();getchar();
	return 0;
}